pkg/network/protocols/events/batch_offsets.go

```diff
@@ -0,0 +1,93 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package events
+
+import (
+	"sync"
+)
+
+// offsetManager is responsible for keeping track of which chunks of data we
+// have consumed from each batch object
+type offsetManager struct {
+	mux        sync.Mutex
+	stateByCPU []*cpuReadState
+}
+
+type cpuReadState struct {
+	// this is the nextBatchID we're expecting for a particular CPU core. we use
+	// this when we attempt to retrieve data that hasn't been sent from kernel space
+	// yet because it belongs to an incomplete batch.
+	nextBatchID int
+
+	// information associated to partial batch reads
+	partialBatchID int
+	partialOffset  int
+}
+
+func newOffsetManager(numCPUS int) *offsetManager {
+	stateByCPU := make([]*cpuReadState, numCPUS)
+	for i := range stateByCPU {
+		stateByCPU[i] = new(cpuReadState)
+	}
+
+	return &offsetManager{stateByCPU: stateByCPU}
+}
+
+// Get returns the data offset that hasn't been consumed yet for a given batch
+func (o *offsetManager) Get(cpu int, batch *batch, syncing bool) (i, j int) {
```

Comment:
IMHO, renaming will ease readability 
```suggestion
func (o *offsetManager) Get(cpu int, batch *batch, syncing bool) (begin, end int) {
```